src/storage/txn/commands/prewrite.rs

```diff
@@ -457,6 +481,16 @@ impl<K: PrewriteKind> Prewriter<K> {
                             .insert(key, (old_value, Some(mutation_type)));
                     }
                 }
+                Err(MvccError(box MvccErrorInner::WriteConflict {
+                    start_ts,
+                    conflict_commit_ts,
+                    ..
+                })) if conflict_commit_ts > start_ts => {
+                    return check_committed_record(prewrite_result, reader, &key);
+                }
+                Err(MvccError(box MvccErrorInner::PessimisticLockNotFound { .. })) => {
+                    return check_committed_record(prewrite_result, reader, &key);
+                }
```

Comment:
It seems that a repeated prewrite will not encounter a false `AlreadyExist`. Before checking data constraint, it will always be rejected by `WriteConflict` first.

Reply:
ah, yes.